### PR TITLE
Initial version of trySubmit with single full error

### LINKED ISSUE #16895: Implement TrySubmit
Part of #16862 
Should not be merged into main daml script, instead we should fork daml script (pending discussion)

Implement `trySubmit : [Party] -> [Party] -> Commands a -> Script (Either SubmitError a)`, as well as implementing `SubmitError` to cover all the cases we can achieve.

Getting errors across ledger API is quite involved, so for now, simply implement in regards to IDELedger and set up the interface in the ScriptLedgerClients.

### PR DESCRIPTION
**Pull request by samuel-williams-da**
WIP
Intention is to add `trySubmit`, which will replace the inner workings of submit and submitMustFail
Factoring out further to `trySubmitTree` is still to be considered, pending limitations with json client.

Currently only fully implemented the Missing Create Authorization exception

Resolves #16895 

TODO:
- Discuss whether this is the interface/error structure we want
- Move over to Daml3.Script pending #16898

### CODE REVIEW

**Comment by samuel-williams-da** [comment]
Next steps - Errors over the IDE are nice and easy, but over the Grpc and json interface are not.
The protobuf for the command service currently uses regular grpc errors, using a mostly static status code and stringified error.
Ideally this error is encoded into the protobuf and returned directly, so we can pull it apart and translate it to daml script errors.
We'll need to propose grpc changes to canton (TODO, talk to @remyhaemmerle-da about this)

**Comment by remyhaemmerle-da** [comment]
Nice !

**Comment by remyhaemmerle-da** [comment]
> Ideally this error is encoded into the protobuf and returned directly, so we can pull it apart and translate it to daml script errors.
> We'll need to propose grpc changes to canton (TODO, talk to @remyhaemmerle-da about this)

I already dig the subject. Unfortunately it seems quite costly to propagate the error properly through the ledger API.   
I think we should start assuming this will not happen on the short terms. 
We may discuss about that during the gathering in Zurich.

**Comment by samuel-williams-da** [comment]
Sure okay, theres a nice middle ground where I can propagate the errors from the IDE ledger, so we can write out integration tests with that, then have some generic error that holds text for the ones over Ledger API, with the idea of "you probably shouldn't use this yet"

File: daml-script/daml/Daml/Script/Error.daml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Script.Error where
+
+import DA.Stack
+
+data SubmitError
```
Comment (akrmn) [inline daml-script/daml/Daml/Script/Error.daml:8]: I think eventually we will want an opaque representation plus extractor functions (similar to `AnyTemplate` + one `fromAnyTemplate` per type) but this is a good start and does a good job of collecting the error states

**Comment by samuel-williams-da** [comment]
Superseded by #17043

**Pull request closed without merging**